File: Data_Stractures/1-Arrays/Left_Rotation/array-left-rotation.cpp
#include <iostream>

int main(void)
{
	int count, left_rotations;
	std::cin >> count >> left_rotations;

	left_rotations %= count;

	int *nums = new int[count];

	for (size_t i = 0; i < count; i++)
		std::cin >> nums[i];

	for (size_t i = 0; i < count; i++)
		std::cout << nums[(i + left_rotations) % count] << ' ';

	return 0;
}

File: Data_Stractures/1-Arrays/Left_Rotation/array-left-rotation.cs
using System;
using System.Linq;

class Program
{
	static void Main(string[] args)
	{
		string[] line = Console.ReadLine().Split(' ');
		int count = Int32.Parse(line[0]);
		int left_rotations = Int32.Parse(line[1]) % count;
		line = Console.ReadLine().Split(' ');
		int[] nums = line.Select(x => Int32.Parse(x)).ToArray();

		for (int i = 0; i < nums.Length; i++)
		{
			Console.Write(nums[(i + left_rotations) % count] + " ");
		}
	}
}
